geopandas/io/tests/test_parquet.py

```diff
@@ -0,0 +1,231 @@
+        _validate_dataframe(df)
+
+    with pytest.raises(ValueError):
+        _validate_dataframe(df.set_index(0))
+
+    # not a DataFrame: should raise ValueError
+    with pytest.raises(ValueError):
+        _validate_dataframe("not a dataframe")
+
+
+@pytest.mark.parametrize(
+    "test_dataset", ["naturalearth_lowres", "naturalearth_cities", "nybb"]
+)
+def test_parquet_io(test_dataset, tmpdir):
+    """Writing to parquet should not raise errors, and should not alter original
+    GeoDataFrame
+    """
+
+    df = read_file(get_path(test_dataset))
+    orig = df.copy()
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+
+    assert os.path.exists(filename)
+
+    # make sure that the original data frame is unaltered
+    assert_geodataframe_equal(df, orig)
+
+    # make sure that we can roundtrip the data frame
+    pq_df = read_parquet(filename)
+
+    assert isinstance(pq_df, GeoDataFrame)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
+
+
+def test_parquet_index(tmpdir):
+    """Setting index=`True` should preserve index in output, and
+    setting index=`False` should drop index from output.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset)).set_index("iso_a3")
+
+    filename = os.path.join(str(tmpdir), "test_with_index.pq")
+    df.to_parquet(filename, index=True)
+    pq_df = read_parquet(filename)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
+
+    filename = os.path.join(str(tmpdir), "drop_index.pq")
+    df.to_parquet(filename, index=False)
+    pq_df = read_parquet(filename)
+    assert_geodataframe_equal(df.reset_index(drop=True), pq_df, check_like=True)
+
+
+@pytest.mark.parametrize("compression", ["snappy", "gzip", "brotli", None])
+def test_parquet_compression(compression, tmpdir):
+    """Using compression options should not raise errors, and should
+    return identical GeoDataFrame.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename, compression=compression)
+    pq_df = read_parquet(filename)
+
+    assert isinstance(pq_df, GeoDataFrame)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
+
+
+def test_parquet_multiple_geom_cols(tmpdir):
+    """If multiple geometry columns are present when written to parquet,
+    they should all be returned as such when read from parquet.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+    df["geom2"] = df.geometry.copy()
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+
+    assert os.path.exists(filename)
+
+    pq_df = read_parquet(filename)
+
+    assert isinstance(pq_df, GeoDataFrame)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
+
+    assert_geoseries_equal(
+        GeoSeries(df.geom2), GeoSeries(pq_df.geom2), check_geom_type=True
```

Comment:
wrapping in GeoSeries *should* not be needed anymore, since nowadays all geometry-dtype columns should be returned as GeoSeries